core/math/math_2d.h

```diff
@@ -336,9 +336,10 @@ struct Rect2 {
 		g.size.height += p_by * 2;
 		return g;
 	}
+
 	inline Rect2 grow_margin(Margin p_margin, real_t p_amount) const {
 		Rect2 g = *this;
-		g.grow_individual((MARGIN_LEFT == p_margin) ? p_amount : 0,
+		g = g.grow_individual((MARGIN_LEFT == p_margin) ? p_amount : 0,
```

Comment:
Isn't the idea that `grow_individual` (just like `grow_margin`) should modify `*this` and return the modified value at the same time? So it should be working already as is, or it's meant to be.

Reply:
I don't know. But as it did not before I didn't change the behaviour.
But for that kind of function, I believe it would make sense that the function modifies the object AND return it.

Reply:
I any case the API is not consistent at all. Some function make a copy, other don't. Some return the modifyed value and other dont. Also I don't get the difference between expand and expand_to ...

Reply:
`Rect2 g = *this;` is a copy, so it won't affect the original one. It is not consistent naming though, since the `grow_*` name implies (IMO) an internal change, not a copy.

We can change the name or the behavior, but either of those is a breaking change, I think it's best to leave as it is (and add this fix, of course).

Reply:
I'll merge this for now, as I need the fix. We can think about how to improve the consistency of the API later :)